doc/configuration.txt

```diff
@@ -1341,6 +1341,7 @@ The following keywords are supported in the "global" section :
    - maxsslconn
    - maxsslrate
    - maxzlibmem
```

Comment:
commit msg:

> This attached patch adds an mptcp option in the config which allows the
> creation of an MPTCP socket instead of TCP on Linux. If Multipath TCP
> is not supported on the system, an error will be reported and the start failed.

That's not correct here: MPTCP is used by default (you need to explain why it is OK to do that here for the server case: you can quote the [website](https://www.mptcp.dev/faq.html#why--when-should-mptcp-be-enabled-by-default) and add a ref to it) if it is supported by the system. If not, TCP is used like before.

You also need to mention:
- `IPPROTO_MPTCP` is defined just in case old libC are being used and don't have the ref. The running kernel is the only one who can tell if MPTCP is supported or not.
- `TCP_MAXSEG` is currently not supported by MPTCP: is it an issue? MPTCP devs didn't add a support for it because it has not been requested with a use-case, but it can be added.